src/runtime/manifest-ast-types/manifest-ast-nodes.ts

```diff
@@ -800,21 +800,110 @@ export interface SchemaAlias extends BaseNode {
 
 export type Expression = ExpressionEntity;
 
+export type PaxelFunctionName = 'now' | 'min' | 'max' | 'average' | 'count' | 'union' | 'first';
+
+interface PaxelFunction {
+  name: PaxelFunctionName;
+  arity: number;
+  returnType: SchemaType;
+}
+
+// represents function(args) => number paxel functions
+function makePaxelNumericFunction(name: PaxelFunctionName, arity: number, type: SchemaPrimitiveTypeValue) {
+  return makePaxelFunction(name, arity, {
+    kind: 'schema-primitive', type, location: INTERNAL_PAXEL_LOCATION
+  });
+}
+
+// Used for builtin function nodes
+const INTERNAL_PAXEL_LOCATION: SourceLocation = {
+  filename: 'internal_paxel_function_table',
+  start: {offset: 0, column: 0, line: 0},
+  end: {offset: 0, column: 0, line: 0}
+};
+
+// Represents function(sequence<type>, ...) => sequence<type> paxel functions
+function makePaxelCollectionTypeFunction(name: PaxelFunctionName, arity: number) {
+  return makePaxelFunction(name, arity, {
+    kind: 'schema-collection',
+    schema: {
+      kind: 'type-name',
+      name: '*', // * denotes a passthrough type, the input type is the same as the output type
+      location: INTERNAL_PAXEL_LOCATION
+    },
+    location: INTERNAL_PAXEL_LOCATION
+  });
+}
+
+// arity = -1 means varargs
+function makePaxelFunction(name: PaxelFunctionName, arity: number, returnType: SchemaType) {
+  return {
+    name,
+    arity,
+    returnType
+  };
+}
+
+export const PAXEL_FUNCTIONS: PaxelFunction[] = [
+  makePaxelNumericFunction('now', 0, 'Number'),
```

Comment:
What does number here mean? Just checking as it could be an Arcs Number or any class that implements Kotlin Number, and we're likely to need to support more than just one type for numeric functions (i.e. generics / typeclasses)

Reply:
Note: src/runtime/refiner.ts already has something similar implemented that you might be able to draw from, but it's admittedly rudimentary.

Reply:
Handling different number types is a bigger task. We will need number casting, overloads or other way to propagate types from arguments to return type. I think it's ok to start with Number and handle the problem of numeric types in a follow up. WDYT?

Reply:
Realistically, we need an inferencer.  In otherwords max(sequence of ints) => int, max(sequence of floats) => float, but it is hard to specify max<T : Number>:T in the current syntax.  I put in a magic type '*', which means foo<T>:T, that is whatever type you take as argument, you return, to use with union(), but we will need something more flexible later.

We could also put smarts into the system elsewhere, so that a 'type tightening' phase could alter this to be the maxType of the inputs as long as the maxType was a subtype of the type specified here (e.g. Number)

So foo(1.2, 3) maxType = Double and foo(3, 4L) = Long, and both are subtypes of Number

Reply:
Seems fine to me, we should probably write up a list of requirements for type inference that the current arcs system doesn't handle. It'd be nice if we could avoid having multiple type checking/inference algorithms (i.e. one for refinements, one for recipes+particles, one for paxel, etc) but I don't know if there's one representation that can handle all of those needs.